test/python/compiler/test_transpiler.py

```diff
@@ -1465,32 +1465,19 @@ def test_target_ideal_gates(self, opt_level):
         theta = Parameter("θ")
         phi = Parameter("ϕ")
         lam = Parameter("λ")
-        target = Target(num_qubits=2)
-        target.add_instruction(UGate(theta, phi, lam))
-        target.add_instruction(CXGate())
-        target.add_instruction(Measure())
+        target = Target(2)
+        target.add_instruction(UGate(theta, phi, lam), {(0,): None, (1,): None})
+        target.add_instruction(CXGate(), {(0, 1): None})
+        target.add_instruction(Measure(), {(0,): None, (1,): None})
         qubit_reg = QuantumRegister(2, name="q")
         clbit_reg = ClassicalRegister(2, name="c")
         qc = QuantumCircuit(qubit_reg, clbit_reg, name="bell")
         qc.h(qubit_reg[0])
         qc.cx(qubit_reg[0], qubit_reg[1])
-        if opt_level != 3:
-            qc.measure(qubit_reg, clbit_reg)
+
         result = transpile(qc, target=target, optimization_level=opt_level)
-        # The Unitary synthesis optimization pass results for optimization level 3
-        # results in a different output than the other optimization levels
-        # and can differ based on fp precision. To avoid relying on a hard match
-        # do a unitary equiv
-
-        if opt_level == 3:
-            result_op = Operator.from_circuit(result)
-            self.assertTrue(result_op.equiv(qc))
-        else:
-            expected = QuantumCircuit(qubit_reg, clbit_reg)
-            expected.u(np.pi / 2, 0, np.pi, qubit_reg[0])
-            expected.cx(qubit_reg[0], qubit_reg[1])
-            expected.measure(qubit_reg, clbit_reg)
-            self.assertEqual(result, expected)
+
+        self.assertEqual(Operator.from_circuit(result), Operator.from_circuit(qc))
```

Comment:
Was there something that prompted this change? Or was it to just make all the optimization levels test the same?

Reply:
yeah I simplified it, we didn't need to make this distinction between the levels in testing.